src/pages/reciters/[reciterId].tsx

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable max-lines */
+import { useState, useMemo } from 'react';
+
+import classNames from 'classnames';
+import Fuse from 'fuse.js';
+import { GetStaticPaths, GetStaticProps } from 'next';
+import useTranslation from 'next-translate/useTranslation';
+import { useRouter } from 'next/router';
+
+import SearchIcon from '../../../public/icons/search.svg';
+import layoutStyle from '../index.module.scss';
+
+import pageStyle from './reciterPage.module.scss';
+
+import { getReciterData } from 'src/api';
+import Footer from 'src/components/dls/Footer/Footer';
+import Input from 'src/components/dls/Forms/Input';
+import NextSeoWrapper from 'src/components/NextSeoWrapper';
+import ChapterList from 'src/components/Reciter/ChapterList';
+import ReciterInfo from 'src/components/Reciter/ReciterInfo';
+import { getAllChaptersData } from 'src/utils/chapter';
+import { logEmptySearchResults } from 'src/utils/eventLogger';
+import { getLanguageAlternates } from 'src/utils/locale';
+import {
+  getCanonicalUrl,
+  getReciterNavigationUrl,
+  getSurahInfoNavigationUrl,
+} from 'src/utils/navigation';
+import Chapter from 'types/Chapter';
+import Reciter from 'types/Reciter';
+
+const filterChapters = (chapters, searchQuery: string) => {
+  const fuse = new Fuse(chapters, {
+    keys: ['transliteratedName'],
+    threshold: 0.3,
+  });
+
+  const filteredReciter = fuse.search(searchQuery);
+  const resultItems = filteredReciter.map(({ item }) => item);
+  if (!filteredReciter.length) {
+    logEmptySearchResults(searchQuery, 'reciter_page_chapter_list');
+  }
+  return resultItems as Chapter[];
+};
+
+type ReciterPageProps = { selectedReciter: Reciter };
+const Reciterpage = ({ selectedReciter }: ReciterPageProps) => {
+  const allChapterData = getAllChaptersData();
+  const router = useRouter();
+  const { t, lang } = useTranslation();
+
+  const reciterId = router.query.reciterId as string;
```

Comment:
reciterId should be passed through the props, not through the router. Check how we do the chapter page Isr